src/compiler/types.ts

```diff
@@ -1473,7 +1473,8 @@ module ts {
         EnumMemberExcludes = Value,
         FunctionExcludes = Value & ~(Function | ValueModule),
         ClassExcludes = (Value | Type) & ~ValueModule,
-        InterfaceExcludes = Type & ~Interface,
+        AmbientClassExcludes = (Value | Type) & ~(ValueModule | Interface),

```

Comment:
If you do the other change I suggested (https://github.com/Microsoft/TypeScript/pull/3333/files#r31552915), then this goes away, and ClassExcludes becomes `(Value | Type) & ~(ValueModule | Interface)`